src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -57,6 +59,14 @@ def _build_x509_name(backend, x509_name):
     return x509.Name(attributes)
 
 
+def _build_general_name(backend, gn):
+    if gn.type == backend._lib.GEN_DNS:
+        data = backend._ffi.buffer(
+            gn.d.dNSName.data, gn.d.dNSName.length
+        )[:].decode("ascii")
+        return x509.DNSName(idna.decode(data))
```

Comment:
So we can drop the `.decode("ascii")` then?

On Tue, Apr 21, 2015 at 9:01 AM, Hynek Schlawack <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1855#discussion_r28774817>:
>
> > @@ -57,6 +59,14 @@ def _build_x509_name(backend, x509_name):
> >      return x509.Name(attributes)
> >
> >
> > +def _build_general_name(backend, gn):
> > +    if gn.type == backend._lib.GEN_DNS:
> > +        data = backend._ffi.buffer(
> > +            gn.d.dNSName.data, gn.d.dNSName.length
> > +        )[:].decode("ascii")
> > +        return x509.DNSName(idna.decode(data))
>
> idna.decode takes bytes so the double decodes are superfluous.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1855/files#r28774817>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084